### PR TITLE
feat: tracker-util to register and track events

### PR DESCRIPTION
**Pull request by paras-fundwave**
Helps fetch event-configs from a store(notion) and register listeners on required DOM-elements to fire tracking events

### CODE REVIEW

File: packages/trackers/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  /**
+   * Fetches events and starts registering observers to begin tracking
+   * note: optionally can pass events, if not they'll be fetched from the store config
+   * @param {IEvent[]} events (optional) - array of events to perform the tracking for
+   **/
+  async initialize(events?: IEvent[]) {
+    if ((window as any).fwTrackersRegistered) return;
+
+    if (!events?.length && !this.store) return console.error("Malformed configuration. Please provide store config or list of events");
```
Comment (Isha-Sharma) [inline packages/trackers/lib/index.ts]: Perhaps, we can throw error?

File: packages/trackers/lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+      this.#debug("Events fetch from store", { events: this.events, store: this.store });
+
+      this.registerTrackers(this.events, this.track);
+      this.observer = new MutationObserver(() => this.registerTrackers(this.events, this.track));
+      this.observer.observe(document, { subtree: true, attributes: true });
+
+      (window as any).fwTrackersRegistered = true;
+    } catch (error) {
+      console.warn("Error while registering event-trackers", error);
```
Comment (Isha-Sharma) [inline packages/trackers/lib/index.ts:51]: Perhaps, we can throw errors?

File: packages/trackers/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "@fundwave/trackers",
```
Comment (Isha-Sharma) [inline packages/trackers/package.json]: Use @fw-components.

File: packages/trackers/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "url": "ssh://git@github.com/getfundwave/custom-elements.git"
+  },
+  "description": "Usage Trackers",
+  "main": "index.js",
+  "scripts": {
+    "prepublishOnly": "npm run build",
+    "build": "tsc --composite false",
+    "dev": "tsc -w --composite false"
+  },
+  "license": "UNLICENSED",
```
Comment (Isha-Sharma) [inline packages/trackers/package.json]: set public use license (MIT)

File: packages/trackers/package.json
Comment (Isha-Sharma) [inline packages/trackers/package.json:1]: Phase 2 - Add test cases.

File: packages/trackers/package.json (hunk 4 above)
Comment (paras-fundwave) [reply packages/trackers/package.json]: added @ [8f6cef5](https://github.com/getfundwave/fw-components/pull/15/commits/8f6cef5ebf8ec6022bb85ac75289f6bbcac7049c)

File: packages/trackers/package.json (hunk 3 above)
Comment (paras-fundwave) [reply packages/trackers/package.json]: updated @ [8f6cef5](https://github.com/getfundwave/fw-components/pull/15/commits/8f6cef5ebf8ec6022bb85ac75289f6bbcac7049c)

File: packages/trackers/lib/index.ts (hunk 1 above)
Comment (paras-fundwave) [reply packages/trackers/lib/index.ts]: wanted to keep this a passive library as this isn't critical so it'll warn instead of throwing errors

Comment (Isha-Sharma) [reply packages/trackers/lib/index.ts]: But the library can still throw it - client can treat it passive - catch and ignore?

**Review by Isha-Sharma — APPROVED** [review]

**Pull request merged**